utils/char16trie/tests/trie_tests.rs

```diff
@@ -0,0 +1,286 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+mod test_util;
+
+use icu_char16trie::char16trie::{Char16Trie, Char16TrieIterator, TrieResult};
+use zerovec::ZeroVec;
+
+#[test]
+fn char16trie_test_empty() {
+    let trie_data = test_util::load_char16trie_data("tests/testdata/empty.toml");
+    let trie = Char16Trie {
+        data: ZeroVec::from_slice(trie_data.as_slice()),
+    };
+
+    let mut itor = Char16TrieIterator::new(trie.data.as_slice());
+    let res = itor.next('h' as i32);
+    assert_eq!(res, TrieResult::NoMatch);
+    assert_eq!(res, TrieResult::NoMatch);
+}
+
+#[test]
+fn char16trie_test_a() {
+    let trie_data = test_util::load_char16trie_data("tests/testdata/test_a.toml");
+    let trie = Char16Trie {
+        data: ZeroVec::from_slice(trie_data.as_slice()),
+    };
+
+    let mut itor = Char16TrieIterator::new(trie.data.as_slice());
+    let res = itor.next('h' as i32);
+    assert_eq!(res, TrieResult::NoMatch);
+
+    let mut itor = Char16TrieIterator::new(trie.data.as_slice());
```

Comment:
If you ported more of the API (e.g., first(unit) and/or reset()) then you could reuse the same object and wouldn't have to reload the test data.

Reply:
This isn't reloading the data, it's just setting a pointer. My expectation for a Rust iterator is that once it is consumed, it won't be reused, which is why I didn't implement `reset`.

Reply:
I also hope we can also do this:

```suggestion
    let mut iter = trie.iter();
```